### PR TITLE
SI-7570 top-level codegen for toolboxes

### PR DESCRIPTION
**Pull request by xeno-by**
Provides a way to inject top-level classes, traits and modules into
toolbox universes.

Previously that was impossible, because compile and eval both wrap their
arguments into an enclosing method of a synthetic module, which makes it
impossible to later on refer to any definitions from the outside.

### CODE REVIEW

File: src/compiler/scala/tools/reflect/ToolBox.scala [hunk 1]
```diff
@@ -102,6 +102,16 @@ trait ToolBox[U <: scala.reflect.api.Universe] {
    */
   def compile(tree: u.Tree): () => Any
 
+  /** Defines a top-level class, trait or module in this ToolBox,
+   *  putting it into a uniquely-named package and returning a symbol that references the defined entity.
+   *  For a ClassDef, a ClassSymbol is returned, and for a ModuleDef, a ModuleSymbol is returned (not a module class, but a module itself).
+   *
+   *  This method can be used to generate definitions that will later be re-used by subsequent calls to
+   *  `compile`, `define` or `eval`. To refer to the generated definition, use the resulting symbol
+   *  by wrapping it into an `Ident`.
```
Comment (densh) [inline src/compiler/scala/tools/reflect/ToolBox.scala]: by splicing it into quasiquote `q"$sym"` ?

Comment (xeno-by) [inline src/compiler/scala/tools/reflect/ToolBox.scala]: fixed

**Comment by xeno-by** [comment]
@retronym Speaking of easy pull requestes, here's one more.

**Comment by retronym** [comment]
LGTM

**Pull request merged**